### PR TITLE
Sheriff drag drop assign

### PR DESCRIPTION
**Pull request by cjam**
Drag and Drop for Sheriffs onto Duties

### CODE REVIEW

File: src/app/components/AssignmentDutyCard/AssignmentDutyCard.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import * as React from 'react'
+import {
+    AssignmentDuty, Sheriff
+} from '../../api/index';
```
Comment (hollyegrayson) [inline src/app/components/AssignmentDutyCard/AssignmentDutyCard.tsx]: Don't need the index

File: src/app/containers/AssignmentDutyDropTarget.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import * as React from 'react'
+import { CSSProperties } from 'react';
+import { 
+    default as dropTargetFactory, 
+    DragDropStatus 
+} from '../infrastructure/DragDrop/dropTargetFactory';
+import ItemTypes from '../infrastructure/DragDrop/ItemTypes';
+import { AssignmentDuty, Sheriff } from '../api';
```
Comment (hollyegrayson) [inline src/app/containers/AssignmentDutyDropTarget.tsx]: different lines?

File: src/app/containers/AssignmentDutyDropTarget.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    computeStyle?: (status: { isActive: boolean, isOver: boolean, canDrop: boolean }) => CSSProperties;
+}
+
+export default class AssignmentDutyDropTarget extends React.PureComponent<AssignmentDutyDropTargetProps> {
+
+    static defaultProps: Partial<AssignmentDutyDropTargetProps> = {
+        canDropItem: (i: DroppableItem) => true
+    }
+
+    // canDropItem(sheriff: DraggedSheriff) {
```
Comment (hollyegrayson) [inline src/app/containers/AssignmentDutyDropTarget.tsx]: Can this be removed?

File: src/app/containers/SheriffDragSource.tsx [hunk 4]
```diff
@@ -1,44 +1,54 @@
...
+        const {children,sheriff} = this.props;
+        return (
+            <GenericSheriffDragSource getDragData={()=>sheriff}>
+                {children}
+            </GenericSheriffDragSource>
+        )
     }
 }
 
+// const mapDispatchToProps = (dispatch: any, ownProps: AssignmentSourceFactoryProps) => {
```
Comment (hollyegrayson) [inline src/app/containers/SheriffDragSource.tsx]: can this be removed?

File: src/app/infrastructure/DragDrop/dropTargetFactory.tsx [hunk 5]
```diff
@@ -41,6 +47,24 @@ export default function dropTargetFactory<TDrag, TDrop>(itemTypes: string | stri
         computeStyle?: (props: { isActive?: boolean, canDrop?: boolean, isOver?: boolean }) => CSSProperties;
     }
 
+    function computeStyleDefault({ isActive, canDrop, isOver }: DragDropStatus) {
+
+        let borderColor: string = 'transparent'
+        //let dropStatus = "";
```
Comment (hollyegrayson) [inline src/app/infrastructure/DragDrop/dropTargetFactory.tsx]: can this be removed?

File: src/app/containers/AssignmentDutyDropTarget.tsx (hunk 3 above)
Comment (cjam) [reply src/app/containers/AssignmentDutyDropTarget.tsx]: Actually didn't need this class anymore anyways, so 👋

File: src/app/infrastructure/DragDrop/dropTargetFactory.tsx (hunk 5 above)
Comment (cjam) [reply src/app/infrastructure/DragDrop/dropTargetFactory.tsx]: Cleaned this up a little more. to match a different way that I had done this.

**Comment by cjam** [comment]
[a. View a operating hours for a given day as 7am to 5pm](https://trello.com/c/HzQIhNCL/161-a-view-a-operating-hours-for-a-given-day-as-7am-to-5pm)

**Comment by cjam** [comment]
[a. Add Sheriffs to a duty](https://trello.com/c/7KXTrntv/162-a-add-sheriffs-to-a-duty)

**Pull request merged**